private_gpt/components/llm/llm_component.py

```diff
@@ -56,3 +56,26 @@ def __init__(self, settings: Settings) -> None:
                 )
             case "mock":
                 self.llm = MockLLM()
+
+            case "bedrock":
+                from llama_index.llms import Bedrock
+
+                self.llm = Bedrock(
+                    model=settings.bedrock.llm_modelid,
+                    temperature=settings.bedrock.llm_temperature
+                )
+
+                from boto3 import Session
+
+                # Access credentials using boto3
+                session = Session()
+                credentials = session.get_credentials()
+
+                # Access key ID and secret access key
+                access_key = credentials.access_key
+                secret_key = credentials.secret_key
+
+                self.llm.set_credentials(aws_region=settings.bedrock.region,
+                                                     aws_access_key_id=access_key,
+                                                     aws_secret_access_key=secret_key
+                                                     )
```

Comment:
Thanks for working on this! I pulled your branch to try it out, and wanted to make a suggestion - it would support more use cases if you created the boto3 client and passed it to `llama_index.llms.Bedrock`, rather than passing in the access key and secret key in `set_credentials`. This allows alternate credential methods like [IRSA](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to work, where there isn't necessarily a credentials file.

```
session = Session()
client = session.client("bedrock-runtime")

self.llm = Bedrock(
    model=settings.bedrock.llm_modelid,
    temperature=settings.bedrock.llm_temperature,
    client=client,
)
```

The same goes for the embeddings model.